doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+        annotation defined:
+
+        .. code-block:: lua
+
+            s:filter(function(w)
+                return w.schema.my_annotation ~= nil
+            end):each(function(w)
+                do_something(w.data)
+            end)
+
+        :param any data: configuration data
+        :param function f: filter function
+
+        :return: a luafun iterator
+
+    ..  _config-schema_object-get:
+
+    ..  method:: get(data, path)
+
+        Get nested configuration values at the given path. The path can be
+        either a dot-separated string (``http.scheme``) or an array-like table (``{ 'http', 'scheme'}``).
+
+        **Example:**
+
+        ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_record_hierarchy/http_api.lua
+            :language: lua
+            :start-at: local scheme
+            :end-before: local host
+            :dedent:
+
+        :param any data: configuration data
+        :param any path: path to the target node
+
+        :return: data at the given path
+
+        **See also**: see :ref:`config_utils_schema_get_configuration`
+
+    ..  _config-schema_object-map:
+
+    ..  method:: map(data, f, f_ctx)
+
+        Transform data by the given function. The ``data`` fields are transformed
+        by the function passed in the second argument (``f``), while its structure remains unchanged.
+
+        The transformation function takes three arguments:
+
+        -   ``data`` -- the configuration data
+        -   ``w`` -- *walkthrough node* with the following fields:
+
+            -   ``w.schema`` -- schema node
+            -   ``w.path`` -- the path to the schema node
+            -   ``w.error()`` -- a function for printing human-readable error messages
+
+        -   ``ctx`` -- additional *context* for the transformation function. Can be
+            used to provide values for a specific call.
+
+        An example of the transformation function:
+
+        .. code-block:: lua
+
+            local function f(data, w, ctx)
+                if w.schema.type == 'string' and data ~= nil then
+                    return data:gsub('{{ *foo *}}', ctx.foo)
+                end
+                return data
+            end
+
+        The ``map()`` method traverses all fields of the schema records,
+        even if they are ``nil`` or ``box.NULL`` in the provided configuration.
+        This allows using this method to set computed default values for missing
+        fields. Note that this is not the case for maps and arrays since the schema
+        doesn't define their fields to traverse.
+
+        :param any data: value at the given path
+        :param any f: walkthrough node, described below
+        :param any f_ctx: user-provided context for the transformation function
+
+        :return: transformed configuration data
+
+    ..  _config-schema_object-merge:
+
+    ..  method:: merge(data_a, data_b)
```

Comment:
I would like to have one important remark here. Quoted from the documentation request:

> A scalar of the `any` type is NOT deeply merged even if it is a table, however it may be useful. We'll consider adding support of such a behaviour in some backward-compatible way in a future.

I want to change it in https://github.com/tarantool/tarantool/issues/10450 (no precise plans here, though), but until that it remains to be the important detail.

Reply:
Added